notebooks/segmented_model_demo/joint_model.py

```diff
@@ -1,18 +1,17 @@
-from typing import List
+from typing import Dict
 
 import numpy as np
 
 
 class JointModel:
-    def __init__(self, segment_models: list, segment_selection_column_idxs: List[int]):
-        self.segment_models = segment_models
-        self.segment_selection_column_idxs = segment_selection_column_idxs
+    def __init__(self, segment_models_idx_mapping: Dict[str,Dict]):
```

Comment:
Again, for genericity, I think `Dict[Any, Tuple[int, IHostedModel]]` and generate this dict from the one the encoder outputs in the notebook (since that's the glueing together of the generic components). For clarity expect an input dict keyed on feature value (`Any` or perhaps `Union[str,int]`) with values which are tuples of (encoded_column_index, segment_model) (`Tuple[int, IHostedModel]`)